nan.h

```diff
@@ -1278,26 +1278,29 @@ class Utf8String {
   class AsyncResource {
    public:
     AsyncResource(
-        MaybeLocal<v8::Object> maybe_resource
-      , v8::Local<v8::String> resource_name) {
+        v8::Local<v8::String> name
+      , v8::Local<v8::Object> resource = New<v8::Object>()) {
 #if NODE_MODULE_VERSION >= NODE_8_0_MODULE_VERSION
       v8::Isolate* isolate = v8::Isolate::GetCurrent();
 
-      v8::Local<v8::Object> resource =
-          maybe_resource.IsEmpty() ? New<v8::Object>()
-                                  : maybe_resource.ToLocalChecked();
+      if (resource.IsEmpty()) {
+        resource = New<v8::Object>();
+      }
```

Comment:
Is this not a tautology?  If `resource` is empty, it is the result of `New<v8::Object>`. Thus, it need not be reinitialized as such. The same would apply to the other constructor.

Reply:
An empty handle is not the same as a handle pointing to a newly allocated object.

```c++
Local<v8::Object> handle1;
Local<v8::Object> handle2 = New<v8::Object>();

assert(handle1.IsEmpty() == true);
assert(handle2.IsEmpty() == false);
```